src/dbnode/storage/cleanup.go

```diff
@@ -283,195 +275,193 @@ func (m *cleanupManager) cleanupExpiredNamespaceDataFiles(earliestToRetain time.
+			// is common for an active commitlog to appear corrupt because the info header has
+			// not been flushed yet.
 			continue
 		}
 
 		m.metrics.corruptCommitlogFile.Inc(1)
 		// If we were unable to read the commit log files info header, then we're forced to assume
 		// that the file is corrupt and remove it. This can happen in situations where M3DB experiences
 		// sudden shutdown.
-		m.opts.InstrumentOptions().Logger().Errorf(
+		m.opts.InstrumentOptions().Logger().WithFields(
+			xlog.NewField("err", errorWithPath.Error()),
+			xlog.NewField("path", errorWithPath.Path()),
+		).Errorf(
 			"encountered corrupt commitlog file during cleanup, marking file for deletion: %s",
 			errorWithPath.Error())
-		filesToCleanup = append(filesToCleanup, newCommitLogFileWithErrorAndPath(
-			commitlog.File{}, errorWithPath.Path(), err))
-	}
-
-	return filesToCleanup, nil
-}
-
-// commitLogNamespaceBlockTimes returns the range of namespace block starts for which the
-// given commit log block may contain data for.
-//
-// consider the situation where we have a single namespace, and a commit log with the following
-// retention options:
-//             buffer past | buffer future | block size
-// namespace:    ns_bp     |     ns_bf     |    ns_bs
-// commit log:     _       |      _        |    cl_bs
-//
-// for the commit log block with start time `t`, we can receive data for a range of namespace
-// blocks depending on the namespace retention options. The range is given by these relationships:
-//	- earliest ns block start = t.Add(-ns_bp).Truncate(ns_bs)
-//  - latest ns block start   = t.Add(cl_bs).Add(ns_bf).Truncate(ns_bs)
-// NB:
-// - blockStart assumed to be aligned to commit log block size
-func commitLogNamespaceBlockTimes(
-	blockStart time.Time,
-	commitlogBlockSize time.Duration,
-	nsRetention retention.Options,
-) (time.Time, time.Time) {
-	earliest := blockStart.
-		Add(-nsRetention.BufferPast()).
-		Truncate(nsRetention.BlockSize())
-	latest := blockStart.
-		Add(commitlogBlockSize).
-		Add(nsRetention.BufferFuture()).
-		Truncate(nsRetention.BlockSize())
-	return earliest, latest
-}
-
-func (m *cleanupManager) cleanupCommitLogs(filesToCleanup []commitLogFileWithErrorAndPath) error {
-	filesToDelete := make([]string, 0, len(filesToCleanup))
-	for _, f := range filesToCleanup {
-		filesToDelete = append(filesToDelete, f.path)
-	}
-	return m.deleteFilesFn(filesToDelete)
-}
-
-type commitLogFileWithErrorAndPath struct {
-	f    commitlog.File
-	path string
-	err  error
-}
-
-func newCommitLogFileWithErrorAndPath(
-	f commitlog.File, path string, err error) commitLogFileWithErrorAndPath {
-	return commitLogFileWithErrorAndPath{
-		f:    f,
-		path: path,
-		err:  err,
-	}
-}
-
-func commitlogsContainPath(commitlogs []commitlog.File, path string) bool {
-	for _, f := range commitlogs {
-		if path == f.FilePath {
-			return true
-		}
+		filesToDelete = append(filesToDelete, errorWithPath.Path())
 	}
 
-	return false
+	return nil
```

Comment:
Hm should this not be `return finalErr` or just `return`?  

Can we add a test that makes sure errors truly come back if the `multiErr` is not nil?

Reply:
This is actually handled by the `defer` statement above which will reset the `finalErr` var to any errors in the `multiErr`. I can change it to `return finalErr` to make it less confusing and I'll make sure we have a test

Reply:
Ok I just verified existing logic was correct but I added a test case for it too